src/motor_control.c

```diff
@@ -64,65 +67,198 @@ void prv_Motor(void *pvParameters) {
+    HWREG(GPIO_PORTB_BASE + GPIO_O_CR)  |= 0xC;
+
+
+    GPIOPinConfigure(x_motor.PWM_Pin_Map);                // configure pin for PWM
+    GPIOPinTypePWM(x_motor.STEP.base, x_motor.STEP.pin);
+
+    /* Count down without synchronization */
+    PWMGenConfigure(x_motor.PWM_Base, X_PWM_BLOCK, PWM_GEN_MODE_DOWN | PWM_GEN_MODE_NO_SYNC);
+
+    /* Set PWM period to: 0.02ms or 50kHz */
+    PWMGenPeriodSet(x_motor.PWM_Base, x_motor.PWM_Block, CALC_PERIOD(PWM0_FREQUENCY));
+
+    /* initialize to no output */
+    PWMPulseWidthSet(x_motor.PWM_Base, X_PWM_OUT, 0);
+
+    /* Enable Interrupts */
+    PWMGenIntRegister(x_motor.PWM_Base, x_motor.PWM_Block, PWM0IntHandler);
+
+    /* Enable the generator block to start timer */
+    PWMGenEnable(x_motor.PWM_Base, x_motor.PWM_Block);
+
+    PWMOutputState(x_motor.PWM_Base, (1 << x_motor.PWM_Channel), true);
+
+}
+
+
+void motor_init_y_pwm(void) {
+
+    y_motor.PWM_Base = Y_MOTOR_PWM_BASE;
+    y_motor.PWM_Channel = Y_MOTOR_PWM_CHANNEL;
+    y_motor.PWM_Block = Y_PWM_BLOCK;
+    y_motor.PWM_Pin_Map = Y_MOTOR_PWM_OUT;
+
     /* setup and enable clock */
     SysCtlPWMClockSet(SYSCTL_PWMDIV_1);                 // Set the PWM clock to the system clock.
 
     // TODO: need to see if there's a way to make this more generic
-    SysCtlPeripheralEnable(SYSCTL_PERIPH_PWM0);         // The PWM peripheral must be enabled for use.
+    SysCtlPeripheralEnable(SYSCTL_PERIPH_PWM1);         // The PWM peripheral must be enabled for use.
 
     /* init GPIO pin */
-    SysCtlPeripheralEnable(motor.STEP.base);            // enable GPIO port if not already enabled
-    GPIOPinConfigure(motor.PWM_Pin_Map);                // configure pin for PWM
-    GPIOPinTypePWM(motor.STEP.base, motor.STEP.pin);
+    /* Should be done in gpio_init */
+    SysCtlPeripheralEnable(y_motor.STEP.base);            // enable GPIO port if not already enabled
+
+    GPIOPinConfigure(y_motor.PWM_Pin_Map);                // configure pin for PWM rather than for GPIO
+    GPIOPinTypePWM(y_motor.STEP.base, y_motor.STEP.pin);
 
     /* Count down without synchronization */
-    PWMGenConfigure(motor.PWM_Base, PWM_GEN_0, PWM_GEN_MODE_DOWN | PWM_GEN_MODE_NO_SYNC);
+    PWMGenConfigure(y_motor.PWM_Base, y_motor.PWM_Block, PWM_GEN_MODE_DOWN | PWM_GEN_MODE_NO_SYNC);
 
     /* Set PWM period to: 0.02ms or 50kHz */
-    PWMGenPeriodSet(motor.PWM_Base, PWM_GEN_0, CALC_PERIOD(PWM0_FREQUENCY));
+    PWMGenPeriodSet(y_motor.PWM_Base, y_motor.PWM_Block, CALC_PERIOD(PWM0_FREQUENCY));
 
     /* initialize to no output */
-    PWMPulseWidthSet(motor.PWM_Base, PWM_OUT_0, 0);
+    PWMPulseWidthSet(y_motor.PWM_Base, Y_PWM_OUT, 0);
+
+    /* Enable Interrupts */
+    PWMGenIntRegister(y_motor.PWM_Base, y_motor.PWM_Block, PWM0IntHandler);
+
+    /* Enable the generator block to start timer */
+    PWMGenEnable(y_motor.PWM_Base, y_motor.PWM_Block);
 }
 
+/* Commented for test */
```

Comment:
use a #ifdef instead of commenting it out

Reply:
Changes made